integration/internal/api/system/system.go

```diff
@@ -0,0 +1,45 @@
+package system
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/internal/test/environment"
+	"github.com/stretchr/testify/require"
+)
+
+// Time provides the current time on the daemon host
+func Time(t *testing.T, client client.APIClient, testEnv *environment.Execution) time.Time {
+	if testEnv.IsLocalDaemon() {
+		return time.Now()
+	}
+
+	ctx := context.Background()
+	info, err := client.Info(ctx)
+	require.Nil(t, err)
+
+	dt, err := time.Parse(time.RFC3339Nano, info.SystemTime)
+	require.Nil(t, err, "invalid time format in GET /info response")
+	return dt
+}
+
+// Version provides the version of the daemon
+func Version(client client.APIClient) (types.Version, error) {
+	ctx := context.Background()
+	return client.ServerVersion(ctx)
+}
+
+// EventsSince returns event and error streams since a provided time
+func EventsSince(client client.APIClient, since string) (<-chan events.Message, <-chan error, func()) {
```

Comment:
This can be an unexported helper next to the one place it is used.

Reply:
It won't be used in one place for long. The `integration-cli` test suite has significant machinery for event stream handling.

Reply:
We should only be testing events from the `integration/system` package, so it doesn't need to be exported.

Reply:
A past authz test author clearly felt that it was necessary to test the interaction between events and authz plugins. In the future, it may be an improvement to move this test elsewhere but in the meantime it seems prudent test the integration between disparate components with the more complicated to test component. Once `integation-cli/` has been eliminated, it should be easy to review all the places where event functionality is used.